### PR TITLE
feat(unit-price) add amount_details attribute to Fee object

### PR DESCRIPTION
**Pull request by ansmonjol**
## Context

We're adding more context to the invoices. PDF have already been updated using the new `amount_details` attribute's values

## Description

This PR maps the new attributes and expose them to GraphQL to have them available for FE

### CODE REVIEW

File: app/graphql/types/fees/amount_details/graduated_percentage_range.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Types
+  module Fees
+    module AmountDetails
+      class GraduatedPercentageRange < Types::BaseObject
+        graphql_name 'FeeAmountDetailsGraduatedPercentageRange'
+
+        field :units, String, null: true
+        field :from_value, Integer, null: true
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/graduated_percentage_range.rb:10]: GraphQL/OrderedFields: Fields should be sorted in an alphabetical order within their section. Field `from_value` should appear before `units`.

File: app/graphql/types/fees/amount_details/graduated_percentage_range.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+        field :to_value, Integer, null: true
+        field :flat_unit_amount, String, null: true
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/graduated_percentage_range.rb:12]: GraphQL/OrderedFields: Fields should be sorted in an alphabetical order within their section. Field `flat_unit_amount` should appear before `to_value`.

File: app/graphql/types/fees/amount_details/graduated_percentage_range.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        field :rate, String, null: true
+        field :per_unit_total_amount, String, null: true
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/graduated_percentage_range.rb:14]: GraphQL/OrderedFields: Fields should be sorted in an alphabetical order within their section. Field `per_unit_total_amount` should appear before `rate`.

File: app/graphql/types/fees/amount_details/graduated_range.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Types
+  module Fees
+    module AmountDetails
+      class GraduatedRange < Types::BaseObject
+        graphql_name 'FeeAmountDetailsGraduatedRange'
+
+        field :units, String, null: true
+        field :from_value, Integer, null: true
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/graduated_range.rb:10]: GraphQL/OrderedFields: Fields should be sorted in an alphabetical order within their section. Field `from_value` should appear before `units`.

File: app/graphql/types/fees/amount_details/object.rb [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+
+        # NOTE: Package charge model
+        field :free_units, String, null: true
+        field :paid_units, String, null: true
+        field :per_package_size, Integer, null: true
+        field :per_package_unit_amount, String, null: true
+
+        # NOTE: Percentage charge model
+        field :units, String, null: true
+        field :free_events, Integer, null: true
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/object.rb:23]: GraphQL/OrderedFields: Fields should be sorted in an alphabetical order within their section. Field `free_events` should appear before `units`.

File: app/graphql/types/fees/amount_details/object.rb [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+        field :rate, String, null: true
+        field :per_unit_total_amount, String, null: true
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/object.rb:25]: GraphQL/OrderedFields: Fields should be sorted in an alphabetical order within their section. Field `per_unit_total_amount` should appear before `rate`.

File: app/graphql/types/fees/amount_details/object.rb [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+        field :paid_events, Integer, null: true
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/object.rb:26]: GraphQL/OrderedFields: Fields should be sorted in an alphabetical order within their section. Field `paid_events` should appear before `per_unit_total_amount`.

File: app/graphql/types/fees/amount_details/object.rb [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        field :fixed_fee_unit_amount, String, null: true
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/object.rb:27]: GraphQL/OrderedFields: Fields should be sorted in an alphabetical order within their section. Field `fixed_fee_unit_amount` should appear before `paid_events`.

File: app/graphql/types/fees/amount_details/object.rb [hunk 9]
```diff
@@ -0,0 +1,36 @@
...
+        field :fixed_fee_total_amount, String, null: true
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/object.rb:28]: GraphQL/OrderedFields: Fields should be sorted in an alphabetical order within their section. Field `fixed_fee_total_amount` should appear before `fixed_fee_unit_amount`.

File: app/graphql/types/fees/amount_details/volume_range.rb [hunk 10]
```diff
@@ -0,0 +1,15 @@
+# frozen_string_literal: true
+
+module Types
+  module Fees
+    module AmountDetails
+      class VolumeRange < Types::BaseObject
+        graphql_name 'FeeAmountDetailsVolumeRange'
+
+        field :per_unit_amount, String, null: false
+        field :flat_unit_amount, String, null: false
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/volume_range.rb:10]: GraphQL/OrderedFields: Fields should be sorted in an alphabetical order within their section. Field `flat_unit_amount` should appear before `per_unit_amount`.

**Review by rsempe — APPROVED** [review]

File: app/graphql/types/fees/amount_details/object.rb [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+
+        # NOTE: Percentage charge model
+        field :fixed_fee_total_amount, String, null: true
+        field :fixed_fee_unit_amount, String, null: true
+        field :free_events, Integer, null: true
+        field :min_max_adjustment_total_amount, String, null: true
+        field :paid_events, Integer, null: true
+        field :rate, String, null: true
+        field :units, String, null: true
+
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/object.rb:27]: Layout/TrailingWhitespace: Trailing whitespace detected. (https://rubystyle.guide#no-trailing-whitespace)

File: app/graphql/types/fees/amount_details/object.rb [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+        field :free_events, Integer, null: true
+        field :min_max_adjustment_total_amount, String, null: true
+        field :paid_events, Integer, null: true
+        field :rate, String, null: true
+        field :units, String, null: true
+        
+        # NOTE: Volume charge model
+        field :flat_unit_amount, String, null: true
+        field :per_unit_amount, String, null: true
+
```
Comment (jdenquin) [inline app/graphql/types/fees/amount_details/object.rb:31]: Layout/TrailingWhitespace: Trailing whitespace detected. (https://rubystyle.guide#no-trailing-whitespace)

**Pull request merged**